src/ast/ast.c

```diff
@@ -571,7 +571,7 @@ cypher_parse_result_t *parse_query(const char *query) {
 		return NULL;
 	}
 
-	AST_RewriteStarProjections(result);
+	if(!AST_ContainsErrors(result)) AST_RewriteStarProjections(result);
```

Comment:
Should we report an error and return NULL?

```c
if(AST_ContainsErrors(result)) {
    _AST_ReportErrors(result);
    return NULL;
}
```
if so we will need to update `ast_validations.c` see `_AST_Validate_ParseResultRoot`

Reply:
`_AST_Validate_ParseResultRoot` is the next function we'll step into, so it feels redundant to me to report errors here. Do you think otherwise?